dev_guide/app_tutorials/maven_tutorial.adoc

```diff
@@ -0,0 +1,166 @@
+[[nexus-introduction]]
+== Introduction
+
+While developing your application with Java and Maven, you will most likely be
+building many times. In order to shorten the build times of your pods, Maven
+dependencies can be cached in a local Nexus repository. This tutorial will
+guide you through creating a Nexus repository on your cluster.
+
+This tutorial assumes that you are working with a project that is already set
+up for use with Maven. If you are interested in using Maven with your Java
+project, it is highly recommended that you look at
+link:https://maven.apache.org/guides/getting-started/index.html[their guide].
+
+In addition, be sure to check your application's image for Maven mirror
+capabilities. Many images that use Maven have a `*MAVEN_MIRROR_URL*` environment
+variable that you can use to simplify this process. If it does not have this
+capability, read
+link:https://books.sonatype.com/nexus-book/reference/config.html[the Nexus documentation]
+to configure your build properly.
+
+[[nexus-setting-up-nexus]]
+== Setting up Nexus
+
+Use `new-app` to download and deploy the official Nexus Docker image.
+
+----
+oc new-app sonatype/nexus
+----
+
+After the build has finished, run:
+
+----
+oc get svc
+----
+
+The output should resemble:
+
+----
+NAME      CLUSTER-IP       EXTERNAL-IP   PORT(S)    AGE
+nexus     172.30.170.249   <none>        8081/TCP   6s
+----
+
+Confirm that Nexus is running by navigating your browser to the new Cluster-IP
+at port 8081. The server may take a bit of time to start, be patient!
+
+Nexus comes pre-configured for the Central Repository, but you may need others
+for your application.  For many Red Hat images, it is recommended to add the
+*jboss-ga* repository at *_https://maven.repository.redhat.com/ga/_*.
+link:https://books.sonatype.com/nexus-book/reference/config-maven.html[Learn how to add repositories].
+The default administrator username is *admin*, and the password is *admin123*.
+
+[[nexus-adding-probes-to-check-for-success]]
+=== Using Probes to Check for Success
+
+This is a good time to set up readiness and liveness probes. These will
+periodically check to see that Nexus is running properly.
+xref:../application_health.adoc#dev-guide-application-health[Learn more about probes].
+
+----
+$ oc set probe dc/nexus \
+	--liveness \
+	--failure-threshold 3 \
+	--initial-delay-seconds 30 \
+	-- echo ok
+$ oc set probe dc/nexus \
+	--readiness \
+	--failure-threshold 3 \
+	--initial-delay-seconds 30 \
+	--get-url=http://nexus:8081/content/groups/public
+----
+
+[[nexus-adding-persistence-to-nexus]]
+=== Adding Persistence to Nexus
+
+[NOTE]
+====
+If you do not want persistent storage, continue to
+xref:nexus-connecting-to-nexus[Connecting to Nexus]. However, your cached
+dependencies will be lost if the pod is restarted for any reason.
+====
+
+Create a persistent volume claim (PVC) for Nexus, so that your dependencies
+will not be lost when the pod running the server terminates. If you do not have
```

Comment:
"Create a persistent volume claim (PVC) for Nexus, so that your dependencies will not be lost when the pod running the server terminates.  PVCs require available persistent volumes (PV) in your cluster.  If there are no PVs available in your cluster and you do not have administrator access....."